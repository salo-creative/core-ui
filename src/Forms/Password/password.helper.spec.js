import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import PasswordHelper from './password.helper';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <PasswordHelper />
  );
  expect(container.firstChild).toMatchSnapshot();
});