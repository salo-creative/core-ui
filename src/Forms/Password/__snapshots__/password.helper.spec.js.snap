// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c3 {
  height: 18px;
  width: 18px;
  margin: 0 0.5rem 0 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: -1rem 0 3rem;
}

.c1 {
  font-size: 1.2rem;
  margin: 0 0 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2rem;
  font-size: 1.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2.valid {
  color: #00ACED;
}

.c2.valid svg path {
  fill: #00ACED;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Your password must contain the following:
  </p>
  <span
    class="c2"
  >
    <svg
      class="c3"
      size="18"
      viewBox="0 0 24 24"
    >
      <path
        d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="#00222b"
      />
    </svg>
    A minimum of 8 characters
  </span>
  <span
    class="c2"
  >
    <svg
      class="c3"
      size="18"
      viewBox="0 0 24 24"
    >
      <path
        d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="#00222b"
      />
    </svg>
    Uppercase letter (A-Z)
  </span>
  <span
    class="c2"
  >
    <svg
      class="c3"
      size="18"
      viewBox="0 0 24 24"
    >
      <path
        d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="#00222b"
      />
    </svg>
    Lowercase letter (a-z)
  </span>
  <span
    class="c2"
  >
    <svg
      class="c3"
      size="18"
      viewBox="0 0 24 24"
    >
      <path
        d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="#00222b"
      />
    </svg>
    Number (0-9)
  </span>
  <span
    class="c2"
  >
    <svg
      class="c3"
      size="18"
      viewBox="0 0 24 24"
    >
      <path
        d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="#00222b"
      />
    </svg>
    Special character (@$!%*?&Â£#)
  </span>
</div>
`;
